test/mount/extensions/autostart_test.clj

```diff
@@ -0,0 +1,34 @@
+(ns mount.extensions.autostart-test
+  (:require [clojure.test :refer [deftest is testing]]
+            [mount.extensions.autostart :as sut]
+            [mount.extensions.explicit-deps :as deps]
+            [mount.lite :as mount]))
+
+(sut/defstate first
+  :dependencies []
+  :start 42)
+
+(sut/defstate second
+  :dependencies []
+  :start 420)
+
+(sut/defstate third
+  :dependencies [#'first #'second]
+  :start (+ @first @second))
+
+(defn fully?
+  [state]
+  (= {#'first state #'second state #'third state} (mount/status)))
+
+(deftest autostart-test
```

Comment:
If I read correctly, I don't think these two `testing` blocks actually test the difference between the default autostart-fn and a custom one? In both cases all three states are started. 

Maybe the `third` defstate should only depend on the `first` defstate, both explicitly as implicitly in the `:start` expression. This way, the default behaviour will start all states, whereas the custom behaviour will only start `first` and `third`.

Reply:
Good catch. Making the `explicit-deps` test case different from the standard `mount/start` case is a better test. Updated